types/mongoose-id-validator/index.d.ts

```diff
@@ -0,0 +1,36 @@
+// Type definitions for mongoose-id-validator 0.6
+// Project: https://github.com/CampbellSoftwareSolutions/mongoose-id-validator
+// Definitions by: Kerollos Magdy <https://github.com/kerolloz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.2
+
+import { Schema, Connection } from 'mongoose';
+
+export interface MongooseIdValidatorOptions {
+    /* Optional custom validation message with {PATH} being replaced
+     * with the relevant schema path that contains an invalid
+     * document ID.
+     */
+    message?: string;
+
+    /* Optional mongoose connection object to use if you are
+     * using multiple connections in your application.
+     *
+     * Defaults to built-in mongoose connection if not specified.
+     */
+    connection?: Connection;
+
+    /* Applies to validation of arrays of ID references only. Set
+     * to true if you sometimes have the same object ID reference
+     * repeated in an array. If set, the validator will use the
+     * total of unique ID references instead of total number of array
+     * entries when checking the database.
+     *
+     * Defaults to false
+     */
+    allowDuplicates?: boolean;
+}
+
+export function mongooseIdValidator(schema: Schema, options?: MongooseIdValidatorOptions): void;
+
+export default mongooseIdValidator;
```

Comment:
Did you mean `exports = mongooseIdValidator`?  That appears to be how the underlying library exports it.

Reply:
The truth is I don't know :sweat_smile:. This is my first time to write Typings for a package. 
I thought it works when I tried it in the tests file. I can easily fix it, please let me know because I am not sure how to do it.

Reply:
There are some notes here: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
TL;DR: Please use `exports = mongooseIdValidator;` in source and `import mongooseIdValidator = require('mongoose-id-validator');` in the tests.

Reply:
![image](https://user-images.githubusercontent.com/36763164/94615495-2bf26800-02a8-11eb-803d-df83b08d5f78.png)
![image](https://user-images.githubusercontent.com/36763164/94615556-4298bf00-02a8-11eb-898e-3164e07ab561.png)
```
Argument of type 'typeof import("/home/codespace/workspace/DefinitelyTyped/types/mongoose-id-validator/index")' is not assignable to parameter of type '(schema: Schema<any>) => void'.
  Type 'typeof import("/home/codespace/workspace/DefinitelyTyped/types/mongoose-id-validator/index")' provides no match for the signature '(schema: Schema<any>): void'. ts(2345)
```

Am I doing something wrong? @amcasey

Reply:
My bad, there's no 's' - it's just `export =`.  Sorry about that.

Reply:
I should also explicitly mention that you'll want to drop the other `export`s, which will require the insertion of `declare` before `function`.

Reply:
Done. :heavy_check_mark:  
Thank you @amcasey.